devito/passes/iet/definitions.py

```diff
@@ -90,6 +90,17 @@ def __init__(self, rcompile=None, sregistry=None, platform=None, **kwargs):
         self.rcompile = rcompile
         self.sregistry = sregistry
         self.platform = platform
+        self.index_mode = kwargs.get('options', {'index-mode': 'int32'})['index-mode']
```

Comment:
we should change the signature into

```
..., platform=Node, options=None, **kwargs):
```

(perhaps at this point the `**kwargs` isn't even necessary anymore)

and then

```
self.index_mode = options['index-mode']
```

if somehow we're calling this without `options` somewhere, we should fix the call site instead